File: testdata/java/Simple.java
import java.util.List;
import java.util.ArrayList;

public interface Greeter {
    String greet(String name);
    String farewell(String name);
}

public class SimpleGreeter implements Greeter {
    private String prefix;

    public SimpleGreeter(String prefix) {
        this.prefix = prefix;
    }

    @Override
    public String greet(String name) {
        return prefix + " Hello, " + name + "!";
    }

    @Override
    public String farewell(String name) {
        return prefix + " Goodbye, " + name + "!";
    }

    public static void main(String[] args) {
        SimpleGreeter greeter = new SimpleGreeter(">>>");
        String result = greeter.greet("world");
        System.out.println(result);
    }
}

File: testdata/go/simple.go
package main

import (
	"fmt"
	"strings"
)

type Greeter interface {
	Greet(name string) string
	Farewell(name string) string
}

type ConsoleGreeter struct {
	Prefix string
}

func (g *ConsoleGreeter) Greet(name string) string {
	return fmt.Sprintf("%s Hello, %s!", g.Prefix, name)
}

func (g *ConsoleGreeter) Farewell(name string) string {
	return fmt.Sprintf("%s Goodbye, %s!", g.Prefix, name)
}

func NewGreeter(prefix string) Greeter {
	return &ConsoleGreeter{Prefix: strings.TrimSpace(prefix)}
}

func main() {
	greeter := NewGreeter(">>>")
	result := greeter.Greet("world")
	fmt.Println(result)
}
